packages/router/__tests__/router-test.ts

```diff
@@ -2306,6 +2306,44 @@ describe("a router", () => {
           child: new Error("Kaboom!"),
         });
       });
+
+      it("clears previous loaderData at that route", async () => {
```

Comment:
Added a handful of new tests because while in here I found that in some cases the router wasn't properly cleaning up stale `actionData`/`loaderData` upon a subsequent error